### PR TITLE
feat: [CDS-76875]: Support for editing remote service

### PR DESCRIPTION
**Pull request by sunnykesh-harness**
### Summary

Changes are : 

- Added support for remote service editing ( opening the git modal, calling update API with applicable params, retaining the payload and other logic similar to Inline flow )
- Fixed redirection for remote service in menu options (edit and referenced by)
- Added timeout to show commit success in gitModal, before just after save it was immediately redirected to details page closing all modal. 

#### Screenshots

https://github.com/harness/harness-core-ui/assets/63278928/5a518e17-9ff3-43c6-ba4d-69eff510128b

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/75-cd/components/PipelineSteps/DeployServiceStep/NewEditServiceModal.tsx [hunk 1]
```diff
@@ -110,7 +110,10 @@ export const NewEditServiceModal: React.FC<NewEditServiceModalProps> = ({
         showSuccess(getString('cd.serviceCreated'))
         // We invalidate the service list call on creating a new service
         queryClient.invalidateQueries(['getServiceAccessList'])
-        payload && onCreateOrUpdate(response?.data?.service)
+        // onCreateOrUpdate will redirect to details page and hide modals showing status
+        setTimeout(() => {
+          response?.data?.service && onCreateOrUpdate(response?.data?.service)
+        }, 1000)
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/components/PipelineSteps/DeployServiceStep/NewEditServiceModal.tsx:116]: Is this timeout of 1s sufficient?

File: src/modules/75-cd/components/Services/ServiceStudio/ServiceStudioDetails.tsx [hunk 2]
```diff
@@ -93,6 +99,91 @@ function ServiceStudioDetails(props: ServiceStudioDetailsProps): React.ReactElem
   const [validateTemplateInputsResponse, setValidateTemplateInputsResponse] =
     React.useState<ResponseValidateTemplateInputsResponseDTO>()
 
+  const getFinalServiceData = useCallback(() => {
+    let updatedService
+    const isVisualView = view === SelectedView.VISUAL
+    const newServiceDefinition = get(pipeline, 'stages[0].stage.spec.serviceConfig.serviceDefinition')
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/components/Services/ServiceStudio/ServiceStudioDetails.tsx:105]: Is this reading serviceDefinition from 1st stage expected? Or we should match the identifier of the service?

File: src/modules/75-cd/components/Services/ServiceStudio/ServiceStudioDetails.tsx [hunk 3]
```diff
@@ -93,6 +99,91 @@ function ServiceStudioDetails(props: ServiceStudioDetailsProps): React.ReactElem
...
+          name: serviceResponse?.name as string
+        })
+      } else {
+        showSuccess(
+          isServiceEntityModalView && isServiceCreateModalView
+            ? getString('common.serviceCreated')
+            : getString('common.serviceUpdated')
+        )
+        fetchPipeline({ forceFetch: true, forceUpdate: true })
+        const newServiceDefinition = get(pipeline, 'stages[0].stage.spec.serviceConfig.serviceDefinition')
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/components/Services/ServiceStudio/ServiceStudioDetails.tsx:151]: Is serviceDefinition always present at 1st stage? We should search for service identifier.

File: src/modules/75-cd/components/Services/ServiceStudio/ServiceStudioDetails.tsx [hunk 4]
```diff
@@ -93,6 +99,91 @@ function ServiceStudioDetails(props: ServiceStudioDetailsProps): React.ReactElem
...
+        setIsDeploymentTypeDisabled?.(!!newServiceDefinition.type)
+      }
+    } else {
+      throw response
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/components/Services/ServiceStudio/ServiceStudioDetails.tsx:155]: Are we catching this response anywhere in top?

**Review by pankajsingh-harness — APPROVED** [review]

File: src/modules/75-cd/components/PipelineSteps/DeployServiceStep/NewEditServiceModal.tsx (hunk 1 above)
Comment (sunnykesh-harness) [reply src/modules/75-cd/components/PipelineSteps/DeployServiceStep/NewEditServiceModal.tsx:116]: Modal with same data and text is already open for 2-3 seconds( till API call starts), after success just loading icon changes to success. And we are closing 1 second after success. Adding more delay will increase latency to user flow. 
Will also discuss this with PM.

File: src/modules/75-cd/components/Services/ServiceStudio/ServiceStudioDetails.tsx (hunk 2 above)
Comment (sunnykesh-harness) [reply src/modules/75-cd/components/Services/ServiceStudio/ServiceStudioDetails.tsx:105]: existing code, just moved to helper function to not repeat.

**Review by chetanvekariya-harness — APPROVED** [review]

**Review by pankajsingh-harness — DISMISSED** [review]

**Review by chetanvekariya-harness — DISMISSED** [review]

**Review by samarth-sharma31 — APPROVED** [review]

**Review by pankajsingh-harness — APPROVED** [review]

**Review by vikrant-harness — APPROVED** [review]

**Pull request merged**